src/objectAdvanced/task5.js

```diff
@@ -0,0 +1,27 @@
+/*
+Topic: object transformation
+a. Create an object person with fields
+  i. name (your chosen value)
+  ii. surname (your chosen value)
+  iii. age (your chosen value)
+  iv. birthDate (your chosen value)
+b. Create an arrow function with one argument (person)
+c. Function returns a new object - transformed person
+  i. Field fullName is combined of person.name and person.surname
+  ii. Field nickname is combined of person.surname and person.age
+*/
+
+const person = {
+    name: `Dainius`,
+    surname: `Brazas`,
+    age: 37,
+    birthday: `19th of April`,
+}
+
+const newPerson = (person) => {
+    const { name, surname, age, fullName=`${name} ${surname}`, nickName=`${surname} ${age}` } = person;
+    const updatedPerson = { fullName, nickName };
```

Comment:
This is a bit hard to read, and you also mix everything together, try separating the logic into steps.
1. Destrcuture the fields;
2. Create the values fullName and nickName
3. Return the new object with needed fields

The mistake here is that you are utilizing default value of object if the key is not provided, which should not be used for new value creation.

```suggestion
    const { name, surname, age } = person;
    const fullName=`${name} ${surname}`;
    const nickName=`${surname} ${age}`;
    const updatedPerson = { fullName, nickName };
```